include/openssl/params.h

```diff
@@ -0,0 +1,106 @@
+int ossl_param_get_long_long(const OSSL_PARAM *p, const char *key,
+                             long long int *val);
+int ossl_param_get_ulong_long(const OSSL_PARAM *p, const char *key,
+                              unsigned long long int *val);
+int ossl_param_get_size_t(const OSSL_PARAM *p, const char *key, size_t *val);
+
+int ossl_param_set_int8(const OSSL_PARAM *p, const char *key,
+                        const int8_t *val);
+int ossl_param_set_uint8(const OSSL_PARAM *p, const char *key,
+                         const uint8_t *val);
+int ossl_param_set_int16(const OSSL_PARAM *p, const char *key,
+                         const int16_t *val);
+int ossl_param_set_uint16(const OSSL_PARAM *p, const char *key,
+                          const uint16_t *val);
+int ossl_param_set_int32(const OSSL_PARAM *p, const char *key,
+                         const int32_t *val);
+int ossl_param_set_uint32(const OSSL_PARAM *p, const char *key,
+                          const uint32_t *val);
+int ossl_param_set_int64(const OSSL_PARAM *p, const char *key,
+                         const int64_t *val);
+int ossl_param_set_uint64(const OSSL_PARAM *p, const char *key,
+                          const uint64_t *val);
+int ossl_param_set_int(const OSSL_PARAM *p, const char *key, const int *val);
+int ossl_param_set_uint(const OSSL_PARAM *p, const char *key,
+                        const unsigned int *val);
+int ossl_param_set_char(const OSSL_PARAM *p, const char *key,
+                        const char *val);
+int ossl_param_set_uchar(const OSSL_PARAM *p, const char *key,
+                         const unsigned char *val);
+int ossl_param_set_short(const OSSL_PARAM *p, const char *key,
+                         const short int *val);
+int ossl_param_set_ushort(const OSSL_PARAM *p, const char *key,
+                          const unsigned short int *val);
+int ossl_param_set_long(const OSSL_PARAM *p, const char *key,
+                        const long int *val);
+int ossl_param_set_ulong(const OSSL_PARAM *p, const char *key,
+                         const unsigned long int *val);
+int ossl_param_set_long_long(const OSSL_PARAM *p, const char *key,
+                             const long long int *val);
+int ossl_param_set_ulong_long(const OSSL_PARAM *p, const char *key,
+                              const unsigned long long int *val);
+int ossl_param_set_size_t(const OSSL_PARAM *p, const char *key,
+                          const size_t *val);
+
+int ossl_param_get_float(const OSSL_PARAM *p, const char *key, float *val);
+int ossl_param_get_double(const OSSL_PARAM *p, const char *key, double *val);
+
+int ossl_param_set_float(const OSSL_PARAM *p, const char *key, const float *val);
+int ossl_param_set_double(const OSSL_PARAM *p, const char *key, const double *val);
+
+int ossl_param_get_BN(const OSSL_PARAM *p, const char *key, BIGNUM **val);
+int ossl_param_set_BN(const OSSL_PARAM *p, const char *key, const BIGNUM *val);
+
+// WIP rubbish follows
```

Comment:
Get rid of //

Reply:
From here to the bottom needs rewriting.  The intention is to (eventually) provide utility macros to allow the definition of OSSL_PARAMS arrays to be easier and structure safe.